### PR TITLE
refactor: [Cache] simplify code of `FileHandler::getItem()`

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Ensure safety nets to `$data['ttl']` and `$data['time']`

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Cache/Handlers/FileHandler.php [hunk 1]
```diff
@@ -229,24 +229,32 @@ public function isSupported(): bool
...
+        if (! isset($data['time']) || ! is_int($data['time'])) {
+            $data['time'] = 0; // offset time is not needed by increment/decrement methods
+        }
+
         if ($data['ttl'] > 0 && Time::now()->getTimestamp() > $data['time'] + $data['ttl']) {
-            // If the file is still there then try to remove it
-            if (is_file($this->path . $filename)) {
-                @unlink($this->path . $filename);
-            }
+            unlink($this->path . $filename);
```
Comment (kenjis) [inline system/Cache/Handlers/FileHandler.php:257]: It seems the existing code is better.

If the file does not exist, e.g., another process has already removed it,
an E_WARNING level error will be generated.
https://www.php.net/manual/en/function.unlink.php#refsect1-function.unlink-description

File: system/Cache/Handlers/FileHandler.php [hunk 2]
```diff
@@ -229,24 +229,32 @@ public function isSupported(): bool
...
+     * @return array<string, mixed>|false
+     * @phpstan-return array{data: mixed, ttl: int, time: int}|false
      */
     protected function getItem(string $filename)
     {
         if (! is_file($this->path . $filename)) {
             return false;
         }
 
+        /** @phpstan-var array{data: mixed, ttl?: int, time?: int}|mixed~array $data */
```
Comment (kenjis) [inline system/Cache/Handlers/FileHandler.php:241]: What is `mixed~array`?

Comment (paulbalandan) [reply system/Cache/Handlers/FileHandler.php:241]: the `mixed` type excluding `array`. this notation is used by phpstan. i think this come from the bitwise `~`

Comment (paulbalandan) [reply system/Cache/Handlers/FileHandler.php]: Removed the inline var for now. It seems on higher rule levels this gives phpdoc-parser errors.

File: system/Cache/Handlers/FileHandler.php [hunk 3]
```diff
@@ -238,15 +239,21 @@ protected function getItem(string $filename)
...
+        if (! is_array($data)) {
+            return false;
+        }
+
+        if (! isset($data['ttl']) || ! is_int($data['ttl'])) {
             return false;
         }
 
+        if (! isset($data['time']) || ! is_int($data['time'])) {
+            $data['time'] = 0; // offset time is not needed by increment/decrement methods
```
Comment (kenjis) [inline system/Cache/Handlers/FileHandler.php:252]: Why not `return false;`?
It seems the data always should have `'time'`.
https://github.com/codeigniter4/CodeIgniter4/blob/1a83ef489fa6e640a6de7ff3e4f6acf58289d5d5/system/Cache/Handlers/FileHandler.php#L94-L98

Comment (paulbalandan) [reply system/Cache/Handlers/FileHandler.php:252]: the increment/decrement may save the data without `time` key.
https://github.com/codeigniter4/CodeIgniter4/blob/1a83ef489fa6e640a6de7ff3e4f6acf58289d5d5/system/Cache/Handlers/FileHandler.php#L150-L154

Comment (kenjis) [reply system/Cache/Handlers/FileHandler.php:252]: The $data is not saved directly.
https://github.com/codeigniter4/CodeIgniter4/blob/1a83ef489fa6e640a6de7ff3e4f6acf58289d5d5/system/Cache/Handlers/FileHandler.php#L161

Comment (paulbalandan) [reply system/Cache/Handlers/FileHandler.php:252]: Oh, you're right.

Comment (paulbalandan) [reply system/Cache/Handlers/FileHandler.php:252]: The use of `$data` there confused me.

Comment (kenjis) [reply system/Cache/Handlers/FileHandler.php:252]: We should avoid reassigning variables whenever possible.

**Review by kenjis — APPROVED** [review]
LGTM!

**Review by michalsn — APPROVED** [review]

**Pull request merged**